x/data/server/utils.go

```diff
@@ -1,50 +0,0 @@
-package server
-
-import (
-	"context"
-
-	gogotypes "github.com/gogo/protobuf/types"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
-	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	"github.com/regen-network/regen-ledger/x/data"
-)
-
-func (s serverImpl) getID(ctx context.Context, iri string) ([]byte, error) {
-	store := sdk.UnwrapSDKContext(ctx).KVStore(s.storeKey)
-	id := s.iriIDTable.GetID(store, []byte(iri))
-	if len(id) == 0 {
-		return nil, status.Errorf(codes.NotFound, "can't find %s", iri)
-	}
-
-	return id, nil
-}
-
-func (s serverImpl) getEntry(ctx context.Context, store sdk.KVStore, id []byte) (*data.ContentEntry, error) {
-
-	dataAnchor, err := s.stateStore.DataAnchorTable().Get(ctx, id)
-	if err != nil {
-		return nil, err
-	}
-
-	timestamp := &gogotypes.Timestamp{
-		Seconds: dataAnchor.Timestamp.Seconds,
-		Nanos:   dataAnchor.Timestamp.Nanos,
-	}
-
-	iri := string(s.iriIDTable.GetValue(store, id))
```

Comment:
This was the only use of `GetValue` using the lookup table but we store the id in each orm table and we can get the IRI from an id and the id from an IRI (i.e. the orm table already supports reverse lookups).

This makes me question whether we even need to hash the data id... Why not use an auto-incrementing id as we do for other orm tables? Is it still necessary to generate our own compact id?

When anchoring data, we anchor using the content hash. In the current implementation, we then generate a compact id for the anchored data and only allow queries by IRI and content hash. Verifying the compact id with a reverse lookup only verifies that we properly generated the id and stored it in the orm table.

Is there another use case? When would we need to verify the compact id?